Formula/vsts-cli.rb

```diff
@@ -0,0 +1,172 @@
+  end
+
+  resource "vsts" do
+    url "https://files.pythonhosted.org/packages/04/02/f2ad99db32d1946ca20ffc9c862bd1807f24d2cef5e97e053d778d4e016a/vsts-0.1.0b2.tar.gz"
+    sha256 "d0d3a4e99282c00c77fd1f63628ce1a9597ef46ebe7661dd7a50b95128969d11"
+  end
+
+  resource "vsts-cli-build" do
+    url "https://files.pythonhosted.org/packages/e4/39/ed4d17dc99cb5b598e9b822afaf1bcb08fadd6f24251213a51b718ca1042/vsts-cli-build-0.1.0b3.tar.gz"
+    sha256 "e81c237b0a6c0b698e6bec8a0ed08d3ad1e10ce626893cb935aefdd776bee963"
+  end
+
+  resource "vsts-cli-build-common" do
+    url "https://files.pythonhosted.org/packages/44/44/c2776c4ac4546a5bff7bb3c75e79bbc052d0c0f90ab68cc1f40f99ea1acc/vsts-cli-build-common-0.1.0b3.tar.gz"
+    sha256 "b5be56a76d1683ee956f6d32035487945fbe5614262611e493855fe5a7a331e1"
+  end
+
+  resource "vsts-cli-code" do
+    url "https://files.pythonhosted.org/packages/03/2f/48d592bee13bf29632f1ff4709d85dd53442cd7f00d4757b7f73f62c1259/vsts-cli-code-0.1.0b3.tar.gz"
+    sha256 "1cbb0b40f41d9ce96824e4cd4bd13e15925894478171098fcf7526dc65165a01"
+  end
+
+  resource "vsts-cli-code-common" do
+    url "https://files.pythonhosted.org/packages/3d/58/f7dbb36526b5f5231ef9bd8d5dbb2329697fe1413f4bc6e40827e799156d/vsts-cli-code-common-0.1.0b3.tar.gz"
+    sha256 "f7068c89506307fc1462c4049c661113f5718983afd74e4a96614675b0ee3ef4"
+  end
+
+  resource "vsts-cli-common" do
+    url "https://files.pythonhosted.org/packages/61/5a/334d99c29888218cf08636335a0a8e52c4efabe0db1b1b0030d2c1d113b3/vsts-cli-common-0.1.0b3.tar.gz"
+    sha256 "2c6be3c85e96f2c8a6eedf767048f0b8fbae48a07d80a43ed3ef1bad313928e0"
+  end
+
+  resource "vsts-cli-team" do
+    url "https://files.pythonhosted.org/packages/65/3f/c68146367e42b87e751b7027a2c36601fe424757f718cfe50064aba4c701/vsts-cli-team-0.1.0b3.tar.gz"
+    sha256 "d36c23c3d052b2efaa1ae2059e771f8cbd2e988275c51aac21deb50c91b4c2ad"
+  end
+
+  resource "vsts-cli-team-common" do
+    url "https://files.pythonhosted.org/packages/6c/2c/9e7695f678ae14f5ae6cc2799280785254dd447382f902887bc048c3af3f/vsts-cli-team-common-0.1.0b3.tar.gz"
+    sha256 "afe42c0daf84943004658e2e48206753f248b7d47b78c8ffa7ed81faa702586f"
+  end
+
+  resource "vsts-cli-work" do
+    url "https://files.pythonhosted.org/packages/ae/d2/27eb61a9035f7ed6886a59058b94fc62bb500e6da0c0f2ffcd8cf32c4592/vsts-cli-work-0.1.0b3.tar.gz"
+    sha256 "5cd87c84f1491c2f165db56b165cdc38df6ca23e11a3cfcf8f5079317d7b69df"
+  end
+
+  resource "vsts-cli-work-common" do
+    url "https://files.pythonhosted.org/packages/00/fe/466bac890e12d6d65d611fa9ddde5336dd07eef7190b4d46aee9ac2366c8/vsts-cli-work-common-0.1.0b3.tar.gz"
+    sha256 "810e655b6e6eb06a2474d098eea839ea29be1edb22e787caa22358e8d5420a0e"
+  end
+
+  def install
+    virtualenv_create(libexec, "python3")
+    virtualenv_install_with_resources
+  end
+
+  test do
+    subcommands = %w[build code configure feedback login logout project work]
+    subcommands.each do |subcommand|
```

Comment:
We need a test that exercises the some of the functionality of the app.

Reply:
Any meaningful functionality of the app depends on authenticating to an existing VSTS account with a personal access token.

The best way to actually add functional tests would be to create a new VSTS account using a dedicated Homebrew email address managed by the Homebrew maintainers. That way, a dedicated access token for the tests could be generated. The token's permissions can be set to read-only, but since they expire, a new token would need to be generated annually.

Reply:
Of try to perform some operation, with invalid authentication, and confirm that it fails as expected (exit code and error message)